debug_toolbar/store.py

```diff
@@ -0,0 +1,90 @@
+import json
+from collections import defaultdict
+
+from django.core.serializers.json import DjangoJSONEncoder
+from django.utils.module_loading import import_string
+
+from debug_toolbar import settings as dt_settings
+
+
+class DebugToolbarJSONEncoder(DjangoJSONEncoder):
+    def default(self, o):
+        try:
+            return super().default(o)
+        except TypeError:
+            return str(o)
+
+
+def serialize(data):
+    return json.dumps(data, cls=DebugToolbarJSONEncoder)
+
+
+def deserialize(data):
+    return json.loads(data)
+
+
+# Record stats in serialized fashion.
+# Remove use of fetching the toolbar as a whole from the store.
+
+
+class BaseStore:
+    config = dt_settings.get_config().copy()
+
+    @classmethod
+    def ids(cls):
+        raise NotImplementedError
+
+    @classmethod
+    def exists(cls, store_id):
+        raise NotImplementedError
+
+    @classmethod
+    def set(cls, store_id):
+        raise NotImplementedError
+
+    @classmethod
+    def delete(cls, store_id):
+        raise NotImplementedError
```

Comment:
should these four be implemented?

1. `ids()`
2. `exists()`
3. `set()`
4. `delete()`

Reply:
Not here. BaseStore is the interface for the API. The subclasses of BaseStore will have different implementations.